### PR TITLE
Benchmarks : Standardize benchmark result store

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
SUMMARY:
 - Add a schema for Benchmark Results store
 - Other minor updates:
      - Fix benchmark_serving nr_qps_pair_ arg logging

TEST PLAN:
Manual local testing
Benchmark Run : https://github.com/neuralmagic/nm-vllm/actions/runs/8239626671

### CODE REVIEW

File: neuralmagic/benchmarks/scripts/benchmark_serving.py [hunk 1]
```diff
@@ -273,31 +274,64 @@ def main(args: argparse.Namespace):
...
+        result[BenchmarkResult.METADATA_KEY_]["num_prompts"] = num_prompts
+        result[BenchmarkResult.METADATA_KEY_]["request_rate"] = \
             request_rate if request_rate < float("inf") else "inf"
-
-        # Merge with benchmark result
-        result_json = {**result_json, **benchmark_result}
+        result[BenchmarkResult.
+               METADATA_KEY_]["duration"] = benchmark_result['duration']
+        result[BenchmarkResult.METADATA_KEY_][
+            "requests_completed"] = benchmark_result['completed']
```
Comment (andy-neuma) [inline neuralmagic/benchmarks/scripts/benchmark_serving.py:292]: it might be better in the long run to hoist the strings out into variables.

**Review by andy-neuma — APPROVED** [review]
cool

File: neuralmagic/benchmarks/scripts/benchmark_serving.py (hunk 1 above)
Comment (varun-sundar-rabindranath) [reply neuralmagic/benchmarks/scripts/benchmark_serving.py]: done.

File: neuralmagic/benchmarks/scripts/benchmark_result.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+from vllm import __version__ as __vllm_version__
+from typing import Optional
+from types import SimpleNamespace
+from pathlib import Path
+from .common import get_benchmarking_context
+from datetime import datetime
+from dataclasses import dataclass, field
+from enum import Enum
+
+# NOTE - PLEASE READ:
```
Comment (andy-neuma) [inline neuralmagic/benchmarks/scripts/benchmark_result.py:16]: will this error or act weird when there is a mismatch?

Comment (varun-sundar-rabindranath) [reply neuralmagic/benchmarks/scripts/benchmark_result.py:16]: No. I added the JSON version for the Elastic Search usecases - So we know what "keys" are guaranteed to exist under some version.

**Pull request merged**